Dockerfile

```diff
@@ -174,127 +149,112 @@ RUN apt-get update -qq \
 
+# Workbench config
 ENV PATH="/opt/workbench/bin_linux64:$PATH" \
     LD_LIBRARY_PATH="/opt/workbench/lib_linux64:$LD_LIBRARY_PATH"
 
-# nipreps/miniconda:py39_4.12.0rc0
-COPY --from=nipreps/miniconda@sha256:5aa4d2bb46e7e56fccf6e93ab3ff765add74e79f96ffa00449504b4869790cb9 /opt/conda /opt/conda
-
-RUN ln -s /opt/conda/etc/profile.d/conda.sh /etc/profile.d/conda.sh && \
-    echo ". /opt/conda/etc/profile.d/conda.sh" >> ~/.bashrc && \
-    echo "conda activate base" >> ~/.bashrc
-
-# Set CPATH for packages relying on compiled libs (e.g. indexed_gzip)
-ENV PATH="/opt/conda/bin:$PATH" \
-    CPATH="/opt/conda/include:$CPATH" \
-    LD_LIBRARY_PATH="/opt/conda/lib:$LD_LIBRARY_PATH" \
-    LANG="C.UTF-8" \
-    LC_ALL="C.UTF-8" \
-    PYTHONNOUSERSITE=1
-
-RUN conda install -y -n base \
-    -c anaconda \
-    -c conda-forge \
-    convert3d=1.3.0 \
-    && sync \
-    && conda clean -afy; sync \
-    && rm -rf ~/.conda ~/.cache/pip/*; sync \
-    && ldconfig
-
-# Unless otherwise specified each process should only use one thread - nipype
-# will handle parallelization
-ENV MKL_NUM_THREADS=1 \
-    OMP_NUM_THREADS=1
-
 # Create a shared $HOME directory
 RUN useradd -m -s /bin/bash -G users fmriprep
 WORKDIR /home/fmriprep
 ENV HOME="/home/fmriprep" \
     LD_LIBRARY_PATH="/usr/lib/x86_64-linux-gnu:$LD_LIBRARY_PATH"
 
-RUN echo ". /opt/conda/etc/profile.d/conda.sh" >> $HOME/.bashrc && \
-    echo "conda activate base" >> $HOME/.bashrc
+COPY --from=micromamba /bin/micromamba /bin/micromamba
+COPY --from=micromamba /opt/conda/envs/fmriprep /opt/conda/envs/fmriprep
 
-RUN /opt/conda/bin/python -m pip install --no-cache-dir --upgrade templateflow
+ENV MAMBA_ROOT_PREFIX="/opt/conda"
+RUN micromamba shell init -s bash && \
+    echo "micromamba activate fmriprep" >> $HOME/.bashrc
+ENV PATH="/opt/conda/envs/fmriprep/bin:$PATH" \
+    CPATH="/opt/conda/envs/fmriprep/include:$CPATH" \
+    LD_LIBRARY_PATH="/opt/conda/envs/fmriprep/lib:$LD_LIBRARY_PATH"
 
 # Precaching atlases
 COPY scripts/fetch_templates.py fetch_templates.py
-
-RUN /opt/conda/bin/python fetch_templates.py && \
+RUN python fetch_templates.py && \
     rm fetch_templates.py && \
     find $HOME/.cache/templateflow -type d -exec chmod go=u {} + && \
     find $HOME/.cache/templateflow -type f -exec chmod go=u {} +
 
+# FSL environment
+ENV LANG="C.UTF-8" \
+    LC_ALL="C.UTF-8" \
+    PYTHONNOUSERSITE=1 \
+    FSLDIR="/opt/conda/envs/fmriprep" \
+    FSLOUTPUTTYPE="NIFTI_GZ" \
+    FSLMULTIFILEQUIT="TRUE" \
+    FSLLOCKDIR="" \
+    FSLMACHINELIST="" \
+    FSLREMOTECALL="" \
+    FSLGECUDAQ="cuda.q"
+
+# Unless otherwise specified each process should only use one thread - nipype
+# will handle parallelization
+ENV MKL_NUM_THREADS=1 \
+    OMP_NUM_THREADS=1
+
 # Installing FMRIPREP
 COPY --from=src /src/fmriprep/dist/*.whl .
-RUN /opt/conda/bin/python -m pip install --no-cache-dir $( ls *.whl )[all]
+RUN pip install --no-cache-dir $( ls *.whl )[telemetry,test]
```

Comment:
Alternatively, we could just install 

```
pip install --no-cache-dir $( ls *.whl )[telemetry]
```
as a base build step, and then create a testing build step that installs the testing dependencies. But that may be excessively complicated for a very minor size improvement.

Reply:
Yeah, been playing with this. It only shaves off 20MB. I think not worth it.